mailpoet/assets/js/src/automation/editor/components/workflow/step-more-menu.tsx

```diff
@@ -1,44 +1,66 @@
-import { useCallback, useState } from 'react';
-import { DropdownMenu } from '@wordpress/components';
+import { useState } from 'react';
+import { Dropdown, DropdownMenu, MenuItem } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
 import { moreVertical, trash } from '@wordpress/icons';
 import { __ } from '@wordpress/i18n';
 import { Hooks } from 'wp-js-hooks';
 import { PremiumModal } from 'common/premium_modal';
 import { Step as StepData } from './types';
+import { storeName } from '../../store';
 
 type Props = {
   step: StepData;
 };
 
+// I do not like this hack.
+// What I want to achieve: With the lazyOnClose method, I want to be able to control to close
+// the menu from within a menuitem.
+// But there must be a better way then storing currentProps like that.
+let currentProps: Dropdown.RenderProps | null = null;
+function setCurrentProps(current) {
+  currentProps = current;
+}
```

Comment:
Hmm, it seems to me that this is needed only because the menu doesn't autoclose anymore. And that seems to be caused by the custom rendering of the dropdown children.

The problem with this "custom" approach seems to be that the rendering differs from what the component does. It also differs visually. Before:

<img width="198" alt="Screenshot 2022-10-31 at 13 49 15" src="https://user-images.githubusercontent.com/141436/199011678-9a52a3a0-9b91-46c3-88bd-9e0343ef7a97.png">

After:

<img width="313" alt="Screenshot 2022-10-31 at 13 50 13" src="https://user-images.githubusercontent.com/141436/199011843-20456b84-6ec0-45a1-85ff-dc8c2642966e.png">

Could we keep using the "controls" of the component, as it is probably intended to be used? I worry that this customization could break some edge cases (indeed, like auto-close) or focus management/accessibility.

I actually ran into the same issue [here](https://github.com/mailpoet/mailpoet/blob/4c65374a0c9a1fc4a3f8a5a7011c4d5dda578148/mailpoet/assets/js/src/automation/listing/components/cells/actions.tsx#L19-L20), example usage [here](https://github.com/mailpoet/mailpoet/blob/4c65374a0c9a1fc4a3f8a5a7011c4d5dda578148/mailpoet/assets/js/src/automation/listing/components/menu/delete.tsx#L17).

So my idea is — could you make that filter process an array of objects such as `{ key, control, slot }`? That would keep using the "controls" attribute and fix this closing issue as well.

(Last side note, I guess we're using underscores in action/filter names but I can recheck them all at once later this week, maybe.)